playbooks/roles/machine/tasks/install_machine_agent_linux.yml

```diff
@@ -56,9 +56,12 @@
 
   - name: Downloading Machine Agent
     get_url:
-      url: '{{ agent_download_url.stdout }}'
+      url: '{{ af_env }}/{{repo}}/{{agent_file}}'
```

Comment:
a good idea would be to use full names like artifactory_url, artifactory_repository;
These values should probably be part of the configuration (parameters) instead of inventory - machine.yaml;

agent_file should be built by using a combination of agent type (machine agent), OS type (linix, win), and agent version, to match the file name that is what you usually get when you download from appdynamics.download site:
machineagent-bundle-64bit-windows-21.9.0.zip or machineagent-bundle-64bit-linux-21.9.0.zip

name: Adjust the Machine agent file name to match the target OS
set_fact:
agent_file: "machineagent-bundle-64bit-windows-{{agent_version}}"
when: ansible_os_family == 'Windows' and agent_type == 'machine'